### PR TITLE
fix(navigation): made structure more dynamic

### LINKED ISSUE #197: BP selector overlap when one business partner
When a user has a single business partner, the disabled selector does not appear correctly:

image

### PR DESCRIPTION
**Pull request by frontendherodk**
This will allow for usernames to be dynamic in size which was not possible before. I've also added different animation to the navigation though it still needs some work including the business partner menu.

Fixes #197 #198 #223

### CODE REVIEW

File: src/app/components/logo-button/logo-button.controller.js [hunk 1]
```diff
@@ -35,13 +35,11 @@ class LogoButtonController {
 
   toggleMobileMenu() {
     this.scnStateHandlerService.dispatch(TOGGLE_MOBILE_MENU);
-    this.scnVisibilityService.toggleBodyClass(this.mobileMenuShown);
   }
 
   hideMobileMenu() {
     if (this.mobileMenuShown) {
       this.scnStateHandlerService.dispatch(HIDE_MOBILE_MENU);
-      this.scnVisibilityService.toggleBodyClass(this.mobileMenuShown);
```
Comment (dami-gg) [inline src/app/components/logo-button/logo-button.controller.js]: Nice, this was really tricky actually :D

Comment (dami-gg) [inline src/app/components/logo-button/logo-button.controller.js]: Nice that we get rid of this, it was really tricky

**Comment by dami-gg** [comment]
In mobile, when closing the left side menu, it's really visible that the grey layout disappears before the menu, which may be happening since a previous iteration, but it's super clear also that the left menu is collapsed and its height is half of the screen. It's hard to explain, I can show you tomorrow :D

**Comment by dami-gg** [comment]
We can't scroll in the left menu on mobile view anymore

**Comment by dami-gg** [comment]
Shouldn't we give some more room to the right of the help button in desktop?

**Comment by dami-gg** [comment]
With a long name like 'Chuck DFasdfafsafsdafsafsa SFAD' the whole layout breaks when clicking the user menu on desktop

image

**Comment by frontendherodk** [comment]
I've fixed the broken layout and worked on the animations.

They are still not perfect but I wouldn't invest more time in it for now.

**Comment by dami-gg** [comment]
That's way better now. I just see the weird tetris movement when switching from mobile to desktop but no actual user will be switching.

**Comment by dami-gg** [comment]
:+1:

**Comment by frontendherodk** [comment]
👍

**Pull request merged**